python-frontend/src/main/java/org/sonar/python/semantic/ProjectLevelSymbolTable.java

```diff
@@ -86,6 +92,12 @@ public void addModule(FileInput fileInput, String packageName, PythonFile python
     globalSymbolsByModuleName.put(fullyQualifiedModuleName, globalSymbols);
     DjangoViewsVisitor djangoViewsVisitor = new DjangoViewsVisitor();
     fileInput.accept(djangoViewsVisitor);
+    clearGlobalSymbolsByFQN();
```

Comment:
We need to clear this in case a file gets added to the symbol table after it's been created.
It's not needed when creating the symbol table during first computation but I felt given the low cost of this operation (`globalSymbolsByFQN` should remain `null` during the whole first computation step anyway) there was no need to distinguish the use cases. 

There is most likely a way to avoid invalidating the whole cache for pure addition operation but I thought this kind of optimization could come at a later time. It could certainly be done with more clarity as well (perhaps directly from the `PythonIndexer` ?) but I don't have a strong opinion on that tbh. Wdyt?

Reply:
I see your point, although I believe adding a new module is always safe for the existing info in the cache (i.e. `globalSymbolsByFQN`), isn't it?
We made sure that symbols references aren't actually "shared" between modules, thanks to `copySymbol` method.

What I would expect us to do is to update  `globalSymbolsByFQN` whenever a file is added after ST has been created. Do you see some other problems in regard to the cache when we add a new module?

Reply:
I believe it's always safe as well, indeed. There might be corner cases for which it would cause problems but it's true I can't come up with one right now.
I'll update the implementation to avoid such unnecessary clear and we'll see if anything comes up to make us rethink this approach.